spec/controllers/veterans_controller_spec.rb

```diff
@@ -0,0 +1,93 @@
+require "rails_helper"
+
+describe VeteransController do
+  describe "#new" do
+    it "makes a new record" do
+      get :new
+      expect(assigns(:veteran)).to be_new_record
+    end
+
+    it "renders the #new template" do
+      get :new
+      expect(response).to render_template(:new)
+    end
+  end
+
+  describe "#create" do
+    let(:veteran_params) do
+      {
+        email: "billybob@email.com",
+        first_name: "Billy",
+        last_name: "Bob"
+      }
+    end
+
+    let(:new_veteran) { Veteran.new(veteran_params) }
+
+    before do
+      allow(Veteran).to receive(:new).and_return(new_veteran)
+    end
+
+    it "makes a new record with the params" do
+      expect(Veteran).to receive(:new).with(veteran_params).and_return(new_veteran)
+      post :create, veteran: veteran_params, format: :html
+    end
+
+    context "when the record saves successfully" do
+      describe "#html" do
+        it "redirects to the action_path" do
+          post :create, veteran: veteran_params, format: :html
+          expect(response).to redirect_to(action_path)
+        end
+
+        it "shows a notice" do
+          post :create, veteran: veteran_params, format: :html
+          expect(flash[:notice]).to include('Thanks for signing up!')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.